src/build-remote/build-remote.cc

```diff
@@ -123,11 +130,18 @@ static int _main(int argc, char * * argv)
                 for (auto & m : machines) {
                     debug("considering building on remote machine '%s'", m.storeUri);
 
-                    if (m.enabled && std::find(m.systemTypes.begin(),
-                            m.systemTypes.end(),
-                            neededSystem) != m.systemTypes.end() &&
-                        m.allSupported(requiredFeatures) &&
-                        m.mandatoryMet(requiredFeatures)) {
+                    if (!m.enabled) {
+                        debug(format("declined remote machine '%1%' because it is disabled") % m.storeUri);
```

Comment:
The `format` function is deprecated, please do:

```suggestion
                        debug("declined remote machine '%s' because it is disabled", storeUri);
```